src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchService.kt

```diff
@@ -13,24 +14,40 @@ private val log = KotlinLogging.logger {}
 class PrisonerSearchService(private val prisonerSearchClient: PrisonerSearchClient) {
   fun lookupPrisoner(barcodeRecipient: BarcodeRecipient) {
     val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
-    try {
-      val matchPrisonersResponse = prisonerSearchClient.matchPrisoners(prisonerSearchRequest)
-      log.info { "Found ${matchPrisonersResponse.matches.size} potential prisoner matches" }
-    } catch (webClientResponseException: WebClientResponseException) {
-      if (webClientResponseException.statusCode == HttpStatus.FORBIDDEN) {
-        log.info { "Client token does not have correct role to call prisoner-search" }
-      } else {
-        log.error { "Failed to call prisoner-search - statusCode ${webClientResponseException.statusCode}" }
+    prisonerSearchClient.matchPrisoners(prisonerSearchRequest)
+      .onErrorResume {
+        with(it) {
+          if (this is WebClientResponseException) {
+            if (statusCode == HttpStatus.FORBIDDEN) {
+              log.info { "Client token does not have correct role to call prisoner-search" }
+            } else {
+              log.error { "Failed to call prisoner-search - statusCode $statusCode" }
+            }
+          } else {
+            log.error("Failed to call prisoner-search", it)
+          }
+        }
+          .let { Mono.empty() }
+      }
+      // In theory this is a possible way to submit the request on a different thread ... but it can't get the auth token because
+      // `userContext` is on the main request thread, so [WebClientConfiguration#addAuthHeaderFilterFunction] results in a NPE
```

Comment:
Some other people have been experimenting with this approach... [this project](https://github.com/ministryofjustice/hmpps-incentives-api/tree/main/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/config) seems to do some stuff around async calls to upstream services. E.g. I think you might be looking for the [AuthTokenFilterFunction](https://github.com/ministryofjustice/hmpps-incentives-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/config/AuthTokenFilterFunction.kt).